Applied-Upstream: commit: 575294f15f790d4c29e656003ba7f57f8e8242d9
Author: David Mohammed <fossfreedom@ubuntu.com>
Last-Update: 2022-08-02
Description: [PATCH] Resolve thumbnail API change #31

---
 panels/background/cc-background-chooser.c | 15 +++++++++++++++
 panels/user-accounts/cc-avatar-chooser.c  | 13 +++++++++++++
 2 files changed, 28 insertions(+)

diff --git a/panels/background/cc-background-chooser.c b/panels/background/cc-background-chooser.c
index e4f06f977..eff8cb880 100644
--- a/panels/background/cc-background-chooser.c
+++ b/panels/background/cc-background-chooser.c
@@ -248,6 +248,7 @@ on_file_chooser_selection_changed_cb (GtkFileChooser               *chooser,
       g_autofree gchar *mime_type = NULL;
       g_autoptr(GFile) file = NULL;
       GtkWidget *preview;
+      GError *error = NULL;
 
       preview = gtk_file_chooser_get_preview_widget (chooser);
 
@@ -263,9 +264,23 @@ on_file_chooser_selection_changed_cb (GtkFileChooser               *chooser,
 
       if (mime_type)
         {
+#if defined(GNOME_DESKTOP_PLATFORM_VERSION) && GNOME_DESKTOP_PLATFORM_VERSION >= 43
+          pixbuf = gnome_desktop_thumbnail_factory_generate_thumbnail (thumbnail_factory,
+                                                                       uri,
+                                                                       mime_type,
+                                                                       NULL,
+                                                                       &error);
+
+          if (error)
+            {
+              g_warning("could not general thumbnail %s (%s) %s\n", uri, mime_type, error->message);
+              g_clear_error(&error);
+            }
+#else
           pixbuf = gnome_desktop_thumbnail_factory_generate_thumbnail (thumbnail_factory,
                                                                        uri,
                                                                        mime_type);
+#endif
         }
 
       gtk_dialog_set_response_sensitive (GTK_DIALOG (chooser),
diff --git a/panels/user-accounts/cc-avatar-chooser.c b/panels/user-accounts/cc-avatar-chooser.c
index 169afbf07..ef5cd356e 100644
--- a/panels/user-accounts/cc-avatar-chooser.c
+++ b/panels/user-accounts/cc-avatar-chooser.c
@@ -168,6 +168,7 @@ update_preview (GtkFileChooser               *chooser,
                 GnomeDesktopThumbnailFactory *thumb_factory)
 {
         g_autofree gchar *uri = NULL;
+        GError *error = NULL;
 
         uri = gtk_file_chooser_get_uri (chooser);
 
@@ -192,9 +193,21 @@ update_preview (GtkFileChooser               *chooser,
                 }
 
                 if (mime_type) {
+#if defined(GNOME_DESKTOP_PLATFORM_VERSION) && GNOME_DESKTOP_PLATFORM_VERSION >= 43
+                        pixbuf = gnome_desktop_thumbnail_factory_generate_thumbnail (thumb_factory,
+                                                                                     uri,
+                                                                                     mime_type,
+                                                                                     NULL,
+                                                                                     &error);
+                       if (error) {
+                                g_warning("could not general thumbnail %s (%s) %s\n", uri, mime_type, error->message);
+                                g_clear_error(&error);
+                       }
+#else
                         pixbuf = gnome_desktop_thumbnail_factory_generate_thumbnail (thumb_factory,
                                                                                      uri,
                                                                                      mime_type);
+#endif
                 }
 
                 gtk_dialog_set_response_sensitive (GTK_DIALOG (chooser),
-- 
2.34.1

